scripts/install/opt/seagate/cortx/motr/bin/motr_mini_prov.py

```diff
@@ -403,30 +405,34 @@ def update_to_file(self, index, url, machine_id, md_disks):
             Conf.set(index, f"server>{machine_id}>cvg[{i}]>m0d[{j}]>md_seg1",f"{md_disk}")
             Conf.save(index)
 
+# populate self.storage_nodes with machine_id for all storage_nodes
+def get_data_nodes(self):
+    machines: Dict[str,Any] = self.nodes
+    storage_nodes: List[str] = []
+    services = Conf.search(self._index, 'node', 'services', Const.SERVICE_MOTR_IO.value)
+    for machine_id in machines.keys():
+       result = [svc for svc in services if machine_id in svc]
+       # skipped controller , HA and server pod
+       if result:
+           storage_nodes.append(machine_id)
+    return storage_nodes
+
 def update_motr_hare_keys(self, nodes):
     # key = machine_id value = node_info
-    for machine_id, node_info in nodes.items():
-        if (node_info['type'] == 'storage_node' or node_info['type'] == 'data_node'):
-            md_disks_lists = get_md_disks_lists(self, node_info)
-            update_to_file(self, self._index_motr_hare, self._url_motr_hare, machine_id, md_disks_lists)
+    for machine_id in self.storage_nodes:
+        node_info = nodes.get(machine_id)
+        md_disks_lists = get_md_disks_lists(self, node_info)
+        update_to_file(self, self._index_motr_hare, self._url_motr_hare, machine_id, md_disks_lists)
 
 def motr_config_k8(self):
     if not verify_libfabric(self):
         raise MotrError(errno.EINVAL, "libfabric is not up.")
 
```

Comment:
If not storage or data nodes, return from here.

Reply:
Done